packages/datagrid/src/components/DefaultHeaderRenderer/MenuDropDownQualityBar.component.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Overlay, Popover, Button } from 'react-bootstrap';
+import { Action } from '@talend/react-components';
+import './MenuDropDownQualityBar.scss';
+
+
+export default function MenuDropDownQualityBar(props) {
+	const [isOpen, setIsOpen] = React.useState(false);
+	const buttonRef = React.useRef();
+	return (<>
+		<Action
+			buttonRef={buttonRef}
+			className={'button-menu-dropdown'}
+			data-feature={props.dataFeature}
+			label={props.label}
+			link
+			hideLabel
+			onClick={() => {
+				setIsOpen(true);
+			}}
+		/>
+		<Overlay
+			onHide={() => {
+				setIsOpen(false);
+			}}
+			placement="bottom"
+			rootClose
+			show={isOpen}
+			target={buttonRef.current}
+		>
+			<Popover id="grid-header-menu">
+				{props.buttonGroup.map((buttonList, keyButtonGroup) => <div key={keyButtonGroup}>
+					{buttonList.map((button, keyButtonList) => <Button
+						key={keyButtonList}
+							className={'button-menu-dropdown__button'}
+						bsStyle="link"
+							data-feature={button.dataFeature}
+							onClick={() => { button.onClick(); setIsOpen(false); }}>
+							{button.label}
+						</Button>
+					)}
+					{(props.buttonGroup.length - 1) != keyButtonGroup && <div class="button-menu-dropdown__divider" />}
+				</div>
+				)}
```

Comment:
This part seems a bit complex to read, i think the composition API could be a bit different.
Maybe with some children or with some intermediates functions